cs/production.go

```diff
@@ -173,22 +175,46 @@ type builtShip struct {
 // produce all items in the production queue
 func (p *production) produce() productionResult {
 	planet := p.planet
-
 	costCalculator := NewCostCalculator()
 	productionResult := productionResult{}
 	available := Cost{Resources: planet.Spec.ResourcesPerYearAvailable}.AddCargoMinerals(planet.Cargo)
 	newQueue := []ProductionQueueItem{}
 	for itemIndex := range planet.ProductionQueue {
 		item := planet.ProductionQueue[itemIndex]
 		maxBuildable := planet.maxBuildable(p.player, item.Type)
+		var err error
 		var cost Cost
 		if item.Type == QueueItemTypeStarbase && planet.Spec.HasStarbase {
-			cost = costCalculator.StarbaseUpgradeCost(planet.Starbase.Tokens[0].design, item.design)
+			cost, err = costCalculator.StarbaseUpgradeCost(p.rules, p.player.TechLevels, p.player.Race.Spec, planet.Starbase.Tokens[0].design, item.design)
+			if err != nil {
+				log.Warn().
```

Comment:
these errors should be returned. producer will need to be updated to return an error. I'm actually surprised it doesn't already since it looks up designs. I'm probably missing a check somewhere. :)